ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/service/population/internal/PopulationServiceImpl.java

```diff
@@ -232,22 +285,103 @@ public void resetPopulation(DataSet dataSet) {
 	 * 
 	 * @param group the group
 	 * @param groupDef the group entity definition
+	 * @param path A Child Context path
 	 */
-	private void addToPopulation(Group group, EntityDefinition groupDef) {
-		for (QName propertyName : group.getPropertyNames()) {
-			EntityDefinition propertyDef = AlignmentUtil.getChild(groupDef, propertyName);
+	private void addToPopulation(Group group, EntityDefinition groupDef, List<ChildContext> path) {
+		Iterable<? extends EntityDefinition> children = entityDefinitionService
+				.getChildren(groupDef);
+		if (children != null && children.iterator().hasNext()) {
+			for (EntityDefinition def : children) {
+				if (groupDef instanceof TypeEntityDefinition)
+					path = def.getPropertyPath();
+				evaluateContext(group, def, path);
+			}
+		}
+		else {
+			evaluateContext(group, groupDef, path);
+		}
+	}
+
+	private void evaluateContext(Group group, EntityDefinition groupDef, List<ChildContext> path) {
 
-			if (propertyDef != null) {
-				Object[] values = group.getProperty(propertyName);
+		if (path == null || path.isEmpty()) {
+			// group or instance at end of path
+			increase(groupDef, 1);
+		}
+		else {
+			ChildContext context = path.get(0);
```

Comment:
Hm, I am starting to see that you changed the way that addToPopulation works by specifying what you want to limit the count to, via the provided EntityDefinition. It would be good if you would reflect that in the method documentation. "the group entity definition" does not tell anything - my first assumption was that it is directly related to the group.